libs/accounts/recovery-phone/src/lib/sms.manager.ts

```diff
@@ -29,12 +29,47 @@ export class SmsManager {
   ) {}
 
   public async phoneNumberLookup(phoneNumber: string) {
-    const result = await this.client.lookups.v2
-      .phoneNumbers(phoneNumber)
-      .fetch({ fields: this.config.extraLookupFields.join(',') });
-    // Calling toJSON converts PhoneNumberInstance into a
-    // object that just holds state and can be serialized.
-    return result.toJSON();
+    try {
+      this.metrics.increment('sms.phoneNumberLookup.start');
+      const result = await this.client.lookups.v2
+        .phoneNumbers(phoneNumber)
+        .fetch({ fields: this.config.extraLookupFields.join(',') });
+
+      this.metrics.increment('sms.phoneNumberLookup.success');
+
+      /**
+       * See https://www.twilio.com/docs/lookup/v2-api/sim-swap for more details
+       * on the sim swap object returned by Twilio. We currently track only
+       * the last sim swap period.
+       */
+      const simSwapPeriod = result.simSwap?.lastSimSwap?.swappedPeriod;
+      if (simSwapPeriod) {
+        this.metrics.increment('sms.phoneNumberLookup.success.simSwap', {
+          period: simSwapPeriod,
+        });
+      }
+      const smsPumpingRisk = result?.smsPumpingRisk?.smsPumpingRiskScore;
+      if (smsPumpingRisk) {
+        this.metrics.histogram(
```

Comment:
Not sure it matters, but I already added something for this in the service layer. Search for `this.metrics.gauge('sim_pumping_risk', smsPumpingRisk);`  

I _think_ maybe the service is the better place to do this since we can control the stat for a lookup during phone number confirm, vs a lookup that might happen in the future during a sign in confirmation.

Reply:
I don't think it makes sense to capture the metric I added, and this metric. I like how you have multiple potential metrics being captured here though... Maybe you should just delete the metric I added? /notsure